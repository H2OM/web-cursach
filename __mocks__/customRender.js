import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}