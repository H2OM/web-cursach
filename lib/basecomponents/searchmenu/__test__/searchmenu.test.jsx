import customRender from "@/__mocks__/customRender";
import SearchMenu from "../searchmenu";
import { screen } from "@testing-library/react";

const context = {
    searchMenu: true,
    searchMenuContent: [
        {
            title: "test1",
            article: "test1article",
            image: "/img/image.jpg"
        },
        {
            title: "test2",
            article: "test2article",
            image: "/img/image.jpg"
        },
        {
            title: "test3",
            article: "test3article",
            image: "/img/image.jpg"
        },
        {
            title: "test4",
            article: "test4article",
            image: "/img/image.jpg"
        },
    ]
}

it('searchmenu render test', async ()=>{
    const {container} = customRender(<SearchMenu/>, {providerProps: context});
    const images = screen.getAllByRole('img');
    expect(images.length).toBe(context.searchMenuContent.length);
    context.searchMenuContent.forEach(each=>{
        expect(screen.getByText(each.title)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();
});