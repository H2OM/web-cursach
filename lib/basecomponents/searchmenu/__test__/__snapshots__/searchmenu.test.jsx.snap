// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchmenu render test 1`] = `
<div>
  <div
    class="Searchmenu Searchmenu_show"
  >
    <div
      class="Searchmenu__list"
    >
      <div
        class="Searchmenu__list__block"
      >
        <a
          href="/catalog/test1article"
        >
          <img
            alt="image"
            class="Favmenu__list__block__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100"
            srcset="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100 2x"
            style="color: transparent;"
            width="300"
          />
        </a>
        test1
      </div>
      <div
        class="Searchmenu__list__block"
      >
        <a
          href="/catalog/test2article"
        >
          <img
            alt="image"
            class="Favmenu__list__block__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100"
            srcset="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100 2x"
            style="color: transparent;"
            width="300"
          />
        </a>
        test2
      </div>
      <div
        class="Searchmenu__list__block"
      >
        <a
          href="/catalog/test3article"
        >
          <img
            alt="image"
            class="Favmenu__list__block__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100"
            srcset="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100 2x"
            style="color: transparent;"
            width="300"
          />
        </a>
        test3
      </div>
      <div
        class="Searchmenu__list__block"
      >
        <a
          href="/catalog/test4article"
        >
          <img
            alt="image"
            class="Favmenu__list__block__image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="300"
            src="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100"
            srcset="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=384&q=100 1x, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage.jpg&w=640&q=100 2x"
            style="color: transparent;"
            width="300"
          />
        </a>
        test4
      </div>
    </div>
  </div>
</div>
`;
